### PR TITLE
Upgrades incremental dom to latest stable version

### PR DESCRIPTION
**Pull request by pragmaticivan**

### CODE REVIEW

**Comment by pragmaticivan** [comment]
This version uses `this` instead `window`. 

Also not clear why we're not using the node module instead copying the file source.

`const IncrementalDOM = require('incremental-dom')`;

File: packages/metal-incremental-dom/src/incremental-dom.js [hunk 1]
```diff
@@ -20,8 +18,10 @@ import { globals } from 'metal';
  */
 
 (function (global, factory) {
-  (factory((global.IncrementalDOM = global.IncrementalDOM || {})));
-}(globals.window, function (exports) { 'use strict';
+  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
```
Comment (mairatma) [inline packages/metal-incremental-dom/src/incremental-dom.js:21]: These lines will probably cause some problems with some build types we support (I believe that for gulp-metal's globals build specifically, or maybe AMD as well). That's why I used to change it manually to the lines you've replaced here:

```js
(function (global, factory) {
   (factory((global.IncrementalDOM = global.IncrementalDOM || {})));
 }(globals.window, function (exports) { 'use strict';
```

File: packages/metal-incremental-dom/src/incremental-dom.js [hunk 2]
```diff
@@ -666,9 +666,7 @@ import { globals } from 'metal';
    * @param {?Object<string, !Element>} keyMap
    */
   var removeChild = function (node, child, keyMap) {
-    if (child.parentNode === node) {
```
Comment (mairatma) [inline packages/metal-incremental-dom/src/incremental-dom.js:669]: This was actually an important fix that they did in incremental-dom. Without it some places throw errors. It's just making sure that `removeChild` won't be called if the given node is not the parent. I don't think we should remove that.

**Comment by mairatma** [comment]
We're not using the node module because it took a long time to be updated, and because the format it exports causes problems with some of our build methods (as I explained inline, specifically because of using `this` instead of `window`).

File: packages/metal-incremental-dom/src/incremental-dom.js (hunk 2 above)
Comment (pragmaticivan) [reply packages/metal-incremental-dom/src/incremental-dom.js:669]: Just saw in your latest commit, in the repo, looks like they haven't published in the CDN yet.

File: packages/metal-incremental-dom/src/incremental-dom.js (hunk 1 above)
Comment (pragmaticivan) [reply packages/metal-incremental-dom/src/incremental-dom.js:21]: @mairatma could you explain why use `window` instead this in this case?

Comment (pragmaticivan) [reply packages/metal-incremental-dom/src/incremental-dom.js:21]: Ahh got it in your last comment. Will close this PR. Thanks.

**Comment by pragmaticivan** [comment]
@mairatma I'm trying to use that dependency with nodejs for some tests, but

**Pull request closed without merging**